examples/connectivity_classes.py

```diff
@@ -0,0 +1,96 @@
+"""
+.. _ex-connectivity-measures:
+
+==============================
+Using the connectivity classes
+==============================
+
+Compute different connectivity measures and then demonstrate
+the utility of the class.
+
+Here we compute the Phase Lag Index (PLI) between all gradiometers and showcase
+how we might interface with the connectivity class.
+"""
+
+# Authors: Adam Li <adam2392@gmail.com>
+#
+# License: BSD (3-clause)
+
+import mne
+from mne_connectivity import spectral_connectivity
+from mne.datasets import sample
+
+# %%
+# Set parameters
+data_path = sample.data_path()
+raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'
+event_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw-eve.fif'
+
+# Setup for reading the raw data
+raw = mne.io.read_raw_fif(raw_fname)
+events = mne.read_events(event_fname)
+
+# Add a bad channel
+raw.info['bads'] += ['MEG 2443']
+
+# Pick MEG gradiometers
+picks = mne.pick_types(raw.info, meg='grad', eeg=False, stim=False, eog=True,
+                       exclude='bads')
+
+# Create epochs for the visual condition
+event_id, tmin, tmax = 3, -0.2, 1.5  # need a long enough epoch for 5 cycles
+epochs = mne.Epochs(raw, events, event_id, tmin, tmax, picks=picks,
+                    baseline=(None, 0), reject=dict(grad=4000e-13, eog=150e-6))
+
+# Compute connectivity for band containing the evoked response.
+# We exclude the baseline period:
+fmin, fmax = 4., 9.
+sfreq = raw.info['sfreq']  # the sampling frequency
+tmin = 0.0  # exclude the baseline period
+epochs.load_data().pick_types(meg='grad')  # just keep MEG and no EOG now
+con = spectral_connectivity(
+    epochs, method='pli', mode='cwt_morlet', sfreq=sfreq, fmin=fmin, fmax=fmax,
+    faverage=False, tmin=tmin, mt_adaptive=False, n_jobs=1)
+
+# %%
+# Now, we can look at different functionalities of the connectivity
+# class returned by `spectral_connectivity`.
+
+# the dimensions of the data corresponding to each axis
+print(con.dims)
+
+# the coordinates for each axis of the data
+print(con.coords)
+
+# the underlying data is stored "raveled"
+print(con.shape)
+
+# however, if one asks for the output dense data, then
+# the shape will show the N by N connectivity
+print(con.get_data(output='dense').shape)
+
+# the number of nodes matches the number of signals
+# used to compute the spectral measure
```

Comment:
I find this framing confusing. "Signals" could also refer to the observations, i.e. epochs. I would prefer to make it explicit, so for example say "electrodes" instead of "signals" or "signals from electrodes" if you must keep signal.